src/sentry/utils/request_cache.py

```diff
@@ -0,0 +1,33 @@
+import threading
+
+from django.core.signals import request_finished
+
+from sentry.models import OrganizationMember
+
+_cache = threading.local()
+
+
+def request_cache(func):
+    def wrapped(*args, **kwargs):
+        if not hasattr(_cache, "items"):
+            _cache.items = {}
+        cache_key = (func, repr(args), repr(kwargs))
+        rv = _cache.items.get(cache_key)
+        if rv is None:
+            rv = func(*args, **kwargs)
+            _cache.items[cache_key] = rv
+        return rv
+
+    return wrapped
+
+
+def clear_cache(**kwargs):
+    _cache.items = {}
+
+
+request_finished.connect(clear_cache)
+
+
+@request_cache
+def get_organization_member(user_id, organization_id):
+    return OrganizationMember.objects.get(user_id=user_id, organization_id=organization_id)
```

Comment:
This probably doesn't belong here, since we should treat this file as the generic place the decorator lives. Maybe just put it in `access.py` where it's used?

Reply:
@wedamija I thought about that but I do think there is value in putting all per-request cached functions in one file so it's clear that the function is cached per-request. Putting them all in one file I think makes that more clear to the user. What do you think?

Reply:
Since this is in `utils` I think it's generally better to avoid doing specific things in here. Think of it more as a library or separate framework. If you want to have a separate place where you put the cached functions that could work, but that probably shouldn't be utils.